### DISCUSSION TITLE
Support OpenCRUD

### DISCUSSION THREAD

**Discussion by agritheory** [geldata/gel#1897] [General]
I am not an expert in OpenCRUD; this is a mostly a question about feasibility. Syntactically the this might be able to be solved by extending the GraphQL operators EdgeDB exposes to include/ alias those that OpenCRUD uses. 

EdgeDB's implementation is less verbose but not much less specific and to me, feels better. But it's not a standard; innovations don't start with standards anyway. 

https://www.opencrud.org/#sec-Queries-Filters

**Comment by 1st1** [comment]
Oh, this looks pretty interesting. I guess we can eventually support a subset of it. How "standard" is OpenCRUD? Are there any other libraries that implement it? Because Prisma is far from being a standard.

**Reply by agritheory** [reply]
I don't know. I am happy to do some research and report back. I don't have good numbers or know how to get good numbers on market share, but I think Prisma's market share is non trivial. For databases, I'd want to look at Fauna, Hasura, PostGraphile, Prisma and any others you'd suggest... there's a Java one but I can't remember the name. There are several CMS-flavored things that offer APIs like this too, but they're not databases per se, not sure if they'd be worth it to include.

**Reply by elprans** [reply]
See #187 for a detailed discussion on GraphQL interface.  TLDR, our current approach is more flexible and matches the functional nature of EdgeQL.  @vpetrovykh may have more thoughts on this.

**Reply by 1st1** [reply]
Thanks @elprans, that's a good find
